src/utils/useIsDesktop.ts

```diff
@@ -0,0 +1,24 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+
+export default function useIsDesktop() {
+  const [isDesktop, setIsDesktop] = useState(true);
+
+  useEffect(() => {
+    const handleResize = () => {
+      const width = window.innerWidth;
+      setIsDesktop(width > 768);
+    };
+
+    handleResize(); // Call the function initially
+
+    window.addEventListener('resize', handleResize); // Add event listener for window resize
+
+    return () => {
+      window.removeEventListener('resize', handleResize); // Clean up the event listener on component unmount
+    };
+  }, []);
```

Comment:
ここのuseEffect第2引数にwindow関連の値を入れてあげると16~19までの項目いらない説ありませんか？

Reply:
できませんでした

```
React Hook useEffect has an unnecessary dependency: 'window.innerWidth'. Either exclude it or remove the dependency array. Outer scope values like 'window.innerWidth' aren't valid dependencies because mutating them doesn't re-render the component.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
```